### DISCUSSION TITLE
Welcome to Super Forms Discussions!

### DISCUSSION THREAD

**Discussion by RensTillmann** [RensTillmann/super-forms#38] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other users of **Super Forms**. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other Super Forms users.

To get started, comment below with an introduction of yourself and tell us about what you do with **Super Forms**.

**Comment by jrameriz** [comment]
Wondering if superforms will ever have a connect to API for maybe JSON request to filter data and output..

**Reply by RensTillmann** [reply]
Hi, not sure what you mean by API, can you explain what you are trying to do? The contact entries are just WordPress posts, so you should be able to use the WordPress core API I guess. Super Forms also has some hooks available if needed.

**Reply by jrameriz** [reply]
oh! I know I can use hooks but I'm a lazy guy and was hoping for something more along the lines of something like this: https://wordpress.org/plugins/cf7-to-api-basic-auth/

**Reply by RensTillmann** [reply]
Hi, there is a "POST" option/setting for super forms already, so that it sends all data as JSON to the specified URL as a POST request in the background while still submitting the form. If this isn't enough you might be able to use Zapier instead. Other than that there is no way of doing it other than using hooks. But we have plans to make a more sophisticated solution to set triggers which should include an option to do "POST" request to any API.

**Comment by githubuser-netizen** [comment]
Hi Rens,

I'm a new user of superforms. I just managed to create my form. Basically I have a wordpress + woocommerce site, and I needed my customers to upload a jpg / pdf before they could buy merch from my site.
 
I'm just a little stuck with what to do next. I have created my superform, but I have no idea what to do with it. Should I use the superform to replace the normal woocommerce checkout form?

**Reply by RensTillmann** [reply]
Hi, Super Forms WooCommerce Checkout Add-on will not replace the default WooCommerce checkout form. All it does is adds the products into the Cart, or into the Cart while directing the user to the Checkout page. I am not sure what you are trying to do here but you have to understand that when a user submits the form, the data is saved as a contact entry within your back-end (including any uploaded files with it). Any E-mails will also be sent if you enabled to do so. After that's done, then the form will add any products (which you have defined in the settings on your form) into the cart. After that it will redirect the user to the checkout page. The rest is handled by WooCommerce as usual.

**Comment by AdamReznicek** [comment]
Hi, Im newbie in Super Forms and I have a question.
Is any way to disable auto scrolling if user click on Next or Prev button in multistep form?

Thanks

**Reply by RensTillmann** [reply]
Yes, edit the multi-part then under "Advanced" you can disable scrolling.

**Comment by AdamReznicek** [comment]
Hi, awesome plugin for multipart forms but i need to track user steps (multipart) via Google Analytics.
Is it possible, or is there any tutorial how to do that?

Many thanks

**Reply by RensTillmann** [reply]
Hi, the only thing possible as of now is to track submissions, https://renstillmann.github.io/super-forms/#/analytics-tracking?id=event-tracking

There is no way to track steps/multi-part progression at this moment. Maybe in the future we have more flexible way of tracking the form.

**Reply by AdamReznicek** [reply]
Thanks for quick reply.
I will do that via custom javascript. Is any event for next or prev step?

**Reply by RensTillmann** [reply]
Hmm, not really, the `SUPER.switchMultipart` is triggered. But what you could track is the event that `window.location.hash` is being changed/updated with the current step in the URL. If you can listen to this you might by able to implement some custom JS that will call analytics event.
